dhall/src/Dhall/Main.hs

```diff
@@ -739,42 +736,34 @@ command (Options {..}) = do
             Data.Text.IO.putStrLn (Dhall.Import.hashExpressionToCode normalizedExpression)
 
         Lint {..} -> do
-            case outputMode of
-                Write -> do
-                    (Header header, expression) <- do
-                        getExpressionAndHeader input
+            originalText <- case input of
+                InputFile file -> Data.Text.IO.readFile file
+                StandardInput  -> Data.Text.IO.getContents
 
-                    let lintedExpression = Dhall.Lint.lint expression
+            (Header header, expression) <- do
+                Dhall.Util.getExpressionAndHeaderFromStdinText censor originalText
 
-                    let doc =   Pretty.pretty header
-                            <>  Dhall.Pretty.prettyCharacterSet characterSet lintedExpression
-
-                    case input of
-                        InputFile file -> writeDocToFile file doc
-
-                        StandardInput -> renderDoc System.IO.stdout doc
-
-                Check -> do
-                    originalText <- case input of
-                        InputFile file -> Data.Text.IO.readFile file
-                        StandardInput  -> Data.Text.IO.getContents
+            let lintedExpression = Dhall.Lint.lint expression
 
-                    let name = case input of
-                            InputFile file -> file
-                            StandardInput  -> "(input)"
+            let doc =   Pretty.pretty header
+                    <>  Dhall.Pretty.prettyCharacterSet characterSet lintedExpression
```

Comment:
I was a bit confused here until I realized that `writeToFile` and `renderDoc` also append newlines…